In [90]:
import pandas as pd
import numpy as np


In [91]:
games = pd.read_csv('Game_Nodes.csv', sep=",")

In [92]:
games.head()

Unnamed: 0                  Black  BlackElo        Date  ECO  \
0           0      Steinitz  Wilhelm         0  1886-01-11  D11   
1           1  Zukertort  Johannes H         0  1886-01-13  C47   
2           2      Steinitz  Wilhelm         0  1886-01-15  D10   
3           3  Zukertort  Johannes H         0  1886-01-18  C67   
4           4      Steinitz  Wilhelm         0  1886-01-20  D10   

                    Event   EventDate  GameNumber  HalfMoves  Moves  \
0  World Championship 1st  1886.01.11           1         92     46   
1  World Championship 1st  1886.01.11           2         92     46   
2  World Championship 1st  1886.01.11           3         93     46   
3  World Championship 1st  1886.01.11           4         78     39   
4  World Championship 1st  1886.01.11           5         63     31   

                        Opening Result  Round Site                  White  \
0  Queen's Gambit Declined Slav  Black      1  USA  Zukertort  Johannes H   
1                  Four Knights  Black      2  USA      Steinitz  Wilhelm   
2  Queen's Gambit Declined Slav  White      3  USA  Zukertort  Johannes H   
3                     Ruy Lopez  Black      4  USA      Steinitz  Wilhelm   
4  Queen's Gambit Declined Slav  White      5  USA  Zukertort  Johannes H   

   WhiteElo  
0         0  
1         0  
2         0  
3         0  
4         0

# Confidence Interval voor Moves

In [93]:
moves = games['Moves']

In [94]:
import scipy.stats as st

In [95]:
st.t.interval(0.90,len(moves)-1,loc=moves.mean(),scale=st.sem(moves))

(42.67440372821062, 44.767116739625635)

In [96]:
st.t.interval(0.95,len(moves)-1,loc=moves.mean(),scale=st.sem(moves))

(42.47343031596106, 44.968090151875195)

In [97]:
st.t.interval(0.99,len(moves)-1,loc=moves.mean(),scale=st.sem(moves))

(42.07981068899629, 45.361709778839966)

In [98]:
st.t.interval(0.9999,len(moves)-1,loc=moves.mean(),scale=st.sem(moves))

(41.23448193793594, 46.207038529900316)

Ik heb het gevoel dat de interval steeds accurater wordt naarmate de confidence hoger wordt. De confidence interval wil zeggen dat de kans x% is dat de waarde van de parameter binnen de interval ligt. Hoe hoger de confidence, hoe kleiner de kans dat de waarde van de parameter buiten de interval ligt.

Bij mij wordt de interval steeds groter naarmate de confidence hoger wordt. Toch blijft de interval niet erg groot en zit er minimaal verschil in de 90% en 99.99%.

## 95% confidence interval op eerste 10% van moves

In [99]:
moves.size

684

In [100]:
movesTenPercent = moves.loc[:moves.size*0.1]

Hier zien we dat de size nu ongeveer 10% is van de totale lijst aan moves.

In [101]:
movesTenPercent.size

69

In [102]:
st.t.interval(0.95,len(movesTenPercent)-1,loc=movesTenPercent.mean(),scale=st.sem(movesTenPercent))

(36.50997161600323, 43.63495592022866)

Hier is goed te zien dat de interval erg toeneemt t.o.v. de 95% confidence interval over de gehele lijst. Bij de gehele lijst was dit tussen de 42.5 en 44.9 en nu 36.5 en 43.63.

# Confidence Interval voor Black Elo > 0

In [103]:
eloAboveZero = games['BlackElo'][games['BlackElo'] > 0]

In [104]:
eloAboveZero.head()

566    2785
567    2785
568    2660
569    2785
570    2660
Name: BlackElo, dtype: int64

In [105]:
st.t.interval(0.90,len(eloAboveZero)-1,loc=eloAboveZero.mean(),scale=st.sem(eloAboveZero))

(2702.243011292193, 2712.163768368824)

In [106]:
st.t.interval(0.95,len(eloAboveZero)-1,loc=eloAboveZero.mean(),scale=st.sem(eloAboveZero))

(2701.27825205047, 2713.128527610547)

In [107]:
st.t.interval(0.99,len(eloAboveZero)-1,loc=eloAboveZero.mean(),scale=st.sem(eloAboveZero))

(2699.369303396654, 2715.0374762643632)

In [108]:
st.t.interval(0.9999,len(eloAboveZero)-1,loc=eloAboveZero.mean(),scale=st.sem(eloAboveZero))

(2695.1494945950953, 2719.2572850659217)

In [109]:
eloAboveZero.max()

2785

In [110]:
eloAboveZero.min()

2660

Ook hier zien we dat de interval niet erg groot is, deze neemt naarmate de confidence hoger wordt uiteraard wel wat toe. De stijging is alleen minimaal als we kijken naar de range die van 2660-2785 loopt en de interval 2695-2719.

## 95% confidence interval op eerste 10% van Black Elo

In [111]:
eloAboveZero.size

118

In [112]:
eloAboveZero.head(120)

566    2785
567    2785
568    2660
569    2785
570    2660
       ... 
679    2705
680    2715
681    2705
682    2715
683    2705
Name: BlackElo, Length: 118, dtype: int64

In [113]:
eloAboveZeroTenPercent = eloAboveZero.loc[566:577]

In [114]:
eloAboveZeroTenPercent.head()

566    2785
567    2785
568    2660
569    2785
570    2660
Name: BlackElo, dtype: int64

In [115]:
eloAboveZeroTenPercent.size

12

In [116]:
st.t.interval(0.95,len(eloAboveZeroTenPercent)-1,loc=eloAboveZeroTenPercent.mean(),scale=st.sem(eloAboveZeroTenPercent))

(2692.020409705077, 2773.812923628256)

In [117]:
st.t.interval(0.95,len(eloAboveZero)-1,loc=eloAboveZero.mean(),scale=st.sem(eloAboveZero))

(2701.27825205047, 2713.128527610547)

Ook hier is goed te zijn dat de interval erg groot is als we alleen de eerste 10% van de lijst pakken. De interval is eerst 2692-2773 en bij de volledige lijst is dat 2701-2713. Dit wil zeggen dat de interval van de eerste 10% niet representatief is voor de volledige lijst.